Add new Task APIs documentation (#973)
moodle/devdocs: docs/apis/subsystems/task/adhoc.md

```php
\core\task\manager::reschedule_or_queue_adhoc_task($task);
```

#### Set the retry limit for a task

> Since Moodle 4.4, the maximum number of retries for a task is **12** by default.

Reviewer: Use a `<Since>` tag here too.

https://github.com/moodle/devdocs/pull/974 makes this a global import.

```suggestion

<Since version="4.4" issueNumber="MDL-79128" />

Since Moodle 4.4, the maximum number of retries for a task is **12** by default.
```

In some situations you may wish to limit the number of times a task is retried. This can be accomplished using the `set_attempts_available()` function before queueing the task, for example:

```php
$task->set_attempts_available(2);
\core\task\manager::queue_adhoc_task($task);
```

> #### Get the remaining available attempts of a task

Reviewer: ```suggestion
#### Get the remaining available attempts of a task

```

> Since Moodle 4.4, the maximum number of retries for a task is **12** by default and will be decreased by 1 each time the task is retried.

Reviewer: ```suggestion
<Since version="4.4" issueNumber="MDL-79128" />

Since Moodle 4.4, the maximum number of retries for a task is **12** by default and will be decreased by 1 each time the task is retried.
```

You can get the remaining available attempts of a task by calling the `get_attempts_available()` function.

```php
$task->get_attempts_available();
```

> #### No-retry tasks

Reviewer: ```suggestion
#### No-retry tasks

```

> Since Moodle 4.4, tasks are retried by default. If you want to disable the retry mechanism for a task, you can override the `retry_until_success()` function and set it to false before queueing the task, for example:

Reviewer: ```suggestion

<Since version="4.4" issueNumber="MDL-79128" />

Since Moodle 4.4, tasks are retried by default. If you want to disable the retry mechanism for a task, you can override the `retry_until_success()` function and set it to false before queueing the task, for example:
```

```php
\core\task\manager::reschedule_or_queue_adhoc_task($task);
```

> #### Set the retry limit for a task

Reviewer: ```suggestion
#### Retrying failing tasks
```

<Since version="4.4" issueNumber="MDL-79128" />

> Since Moodle 4.4, the maximum number of retries for a task is **12** by default.

Reviewer: Again:

The number of retries for a failing task can be controlled using the `set_attempts_avaiable` method on the task before it is queued. If not specified a task will be allowed 12 retries.

In some situations you may wish to limit the number of times a task is retried. This can be accomplished using the `set_attempts_available()` function before queueing the task, for example:

```php
$task->set_attempts_available(2);
\core\task\manager::queue_adhoc_task($task);
```

> 
> #### Get the remaining available attempts of a task
> 
> <Since version="4.4" issueNumber="MDL-79128" />

Reviewer: This title has no real benefit. The feature is setting/getting the count.

```suggestion
```

Since Moodle 4.4, the maximum number of retries for a task is **12** by default and will be decreased by 1 each time the task is retried.

You can get the remaining available attempts of a task by calling the `get_attempts_available()` function.

```php
$task->get_attempts_available();
```

> #### No-retry tasks
> 
> <Since version="4.4" issueNumber="MDL-79128" />
> 

Reviewer: This title has no real benefit. The feature is setting/getting the count. This is a part of that feature, not a feature in its own right.

```suggestion
```

> Since Moodle 4.4, tasks are retried by default. If you want to disable the retry mechanism for a task, you can override the `retry_until_success()` function and set it to false before queueing the task, for example:

Reviewer: ```suggestion
Tasks are retried by default, but this behaviour can be modified by overriding the `retry_until_success()` method in your task class, for example:
```
